### ISSUE TITLE
Should receiver name be an underscore?

### ISSUE THREAD

**Issue by rcorre** [golang/lint#310]
I'm trying to implement a `NotFoundHandler` of type `http.Handler`, which has the following interface:

```go
type Handler interface {
	ServeHTTP(ResponseWriter, *Request)
}
```

My implementation looks like the following:

```go
type notFoundHandler struct{}
func (_ notFoundHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
	log.Info("NotFound: %+v", req)
	http.NotFound(resp, req)
}
```

As you can see, I could stand to use a simple function, but I am required to implement an interface.
In this case, my receiver is unused, and I believe the canonical go approach is to use a [blank identifier](https://golang.org/doc/effective_go.html#blank). However, go lint complains:

```
receiver name should not be an underscore
```

Is it really inappropriate to do this? What is the reasoning?

**Comment by dsnet** [comment]
You are allowed to just elide the receiver name entirely in that case.

```go
func (notFoundHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request)
```

We could make the error nicer, though.

**Comment by rcorre** [comment]
Thanks! I think I've seen that before but I'm not sure if its documented anywhere. Good to know. A nicer message would definitely help.
